### DISCUSSION TITLE
What is the best practice for alerting on usage, and/or capping costs?

### DISCUSSION THREAD

**Discussion by david-earnbetter** [deepgram/community#759] [General help] [answered]
I see that there are a few billing-related APIs I could use to build my own cost threshold / anomaly detection, but I wonder if I'm missing something. Are there built-in solutions or known best practices for getting alerts if costs exceed specified thresholds for some unit of time? For capping usage once a cost threshold is reached?

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by davidvonthenen** [comment, marked as answer]
This might help you with your answer as discussed in Discord:
https://discord.com/channels/1108042150941294664/1238494698788884522
